kornia/augmentation/container/augment.py

```diff
@@ -178,36 +191,62 @@ def __init__(
+        return super().clear_state()
+
+    @property
+    def transform_matrix(self) -> Optional[Tensor]:
+        # In AugmentationSequential, the parent class is accessed first.
+        # So that it was None in the begining. We hereby use lazy computation here.
+        if self._transform_matrix is None and len(self._transform_matrices) != 0:
+            self._transform_matrix = self._transform_matrices[0]
+            for mat in self._transform_matrices[1:]:
+                self._update_transform_matrix(mat)
+        return self._transform_matrix
+
+    def _update_transform_matrix_by_module(self, module: Module) -> None:
+        if self._transformation_matrix_arg == "skip":
+            return
+        if isinstance(module, (RigidAffineAugmentationBase2D, AugmentationSequential)):
+            self._transform_matrices.append(module.transform_matrix)
+            # self._update_transform_matrix(module.transform_matrix)
+        elif self._transformation_matrix_arg == "rigid":
+            raise RuntimeError(
+                f"Non-rigid module `{module}` is not supported under `rigid` compuatation mode. "
+                "Please either update the module or change the `transformation_matrix` argument."
+            )
+
+    def _update_transform_matrix(self, transform_matrix: Tensor) -> None:
+        if self._transform_matrix is None:
+            self._transform_matrix = transform_matrix
+        else:
+            self._transform_matrix = transform_matrix @ self._transform_matrix
+
     def identity_matrix(self, input: Tensor) -> Tensor:
         """Return identity matrix."""
         if self.contains_3d_augmentation:
             return eye_like(4, input)
         else:
             return eye_like(3, input)
 
-    @property
-    def transform_matrix(self) -> Optional[Tensor]:
-        return self._transform_matrix
-
     def inverse(  # type: ignore[override]
         self,
-        *args: Tensor,
+        *args: DataType,
         params: Optional[List[ParamItem]] = None,
         data_keys: Optional[List[Union[str, int, DataKey]]] = None,
-    ) -> Union[Tensor, List[Tensor], List[Union[Tensor, List[Tensor]]]]:
+    ) -> Union[DataType, List[DataType]]:
         """Reverse the transformation applied.
 
         Number of input tensors must align with the number of``data_keys``. If ``data_keys`` is not set, use
         ``self.data_keys`` by default.
         """
-        if data_keys is None:
-            _data_keys = self.data_keys
-        else:
-            _data_keys = [DataKey.get(inp) for inp in data_keys]
+        self.transform_op.data_keys = self.transform_op.preproc_datakeys(data_keys)
 
-        self._validate_args_datakeys(*args, data_keys=_data_keys)
+        self._validate_args_datakeys(*args, data_keys=self.transform_op.data_keys)
 
-        args = self._arguments_preproc(*args, data_keys=_data_keys)
+        in_args = self._arguments_preproc(*args, data_keys=self.transform_op.data_keys)
```

Comment:
I think this isn't used anywhere here, so this can be:
```suggestion
        outputs = self._arguments_preproc(*args, data_keys=self.transform_op.data_keys)
```

Reply:
It is used to retain the original input format.